pkg/insights/insightsclient/insightsclient.go

```diff
@@ -66,6 +73,49 @@ func New(client *http.Client, maxBytes int64, metricsName string, authorizer Aut
 	}
 }
 
+func (c *Client) clientTransport() http.RoundTripper {
+	// default transport, proxy from env
+	clientTransport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout: 10 * time.Second,
+		DisableKeepAlives:   true,
+	}
+
+	// get the cluster proxy configuration and override the client transport
+	proxyStatus, err := c.GetProxyStatusFunc()
+	if err != nil {
+		klog.Errorf("Failed to get proxy configuration: %v", err)
+		return transport.DebugWrappers(clientTransport)
+	}
+	if proxyStatus == nil {
+		return transport.DebugWrappers(clientTransport)
+	}
+	clientTransport.Proxy = func(req *http.Request) (*url.URL, error) {
```

Comment:
You wouldn't need to be getting the proxy object and only have your operator injected with the vars by the CVO if you use this annotation: https://github.com/openshift/cluster-authentication-operator/blob/master/manifests/07_deployment.yaml#L9

That would restart your operator on config change. I'll let you pick whichever you like better.